app/elastic_search_indexes/search/goods_nomenclature_index.rb

```diff
@@ -0,0 +1,148 @@
+module Search
+  class GoodsNomenclatureIndex < ::SearchIndex
+    def dataset
+      TimeMachine.now do
+        GoodsNomenclature.actual
+      end
+    end
+
+    def eager_load_graph
+      %i[
+        goods_nomenclature_indents
+        chapter
+        heading
+        ancestors
+      ]
+    end
+
+    def definition
+      {
+        settings: {
+          index: {
+            analysis: {
+              analyzer: {
+                english_exact: {
+                  tokenizer: 'standard',
+                  filter: %w[lowercase],
+                },
+                english: {
+                  tokenizer: 'standard',
+                  filter: %w[
+                    english_possessive_stemmer
+                    lowercase
+                    english_stop
+                    english_stemmer
```

Comment:
my understanding is that we should use a lemmatizer instead of a stemmer, which is rough  and can include similar terms but with different meaning.